apps/rig_auth/lib/rig_auth/jwt/TokenUtils.ex

```diff
@@ -0,0 +1,52 @@
+defmodule RigAuth.Jwt.TokenUtils do
+  use Joken.Config, default_signer: nil # no default signer. Do not set as that will break Jwks retrieval
+
+  use Rig.Config, [:secret_key, :rotation_flag, :jwks_endpoint]
+
+  @conf_var config()
+
+  def config_retrieval do
+    @conf_var
+  end
+
+  if conf_var.rotation_flag do
```

Comment:
similar to above, this code is executed when the module is compiled, which might not be what you want